crates/libs/core/src/runtime/stateful.rs

```diff
@@ -43,20 +43,28 @@ pub trait LocalStatefulServiceReplica: Send + Sync + 'static {
     fn abort(&self);
 }
 
+#[derive(Debug, Clone)]
 pub struct StatefulServicePartition {
-    _com_impl: IFabricStatefulServicePartition,
+    com_impl: IFabricStatefulServicePartition,
 }
 
 impl StatefulServicePartition {
     pub fn get_com(&self) -> &IFabricStatefulServicePartition {
-        &self._com_impl
+        &self.com_impl
+    }
+
+    /// Reports load for the current replica in the partition.
+    pub fn report_load(&self, metrics: &[LoadMetric]) -> crate::Result<()> {
```

Comment:
To confirm my understanding, the application itself periodically calls `report_load()` itself to report load information. SF itself does not call this as a callback, but instead the application is responsible for calling this on some cadence

Reply:
Yes. Application is responsible for calling this API. If app does not report load, the default load in the app/service manifest will be used by SF. App reported load will replace the default load, and the latest reported load will be used by SF for balancing decisions.